core_v2/sources/v2_config.move

```diff
@@ -0,0 +1,388 @@
+/*
+Provides a singleton wrapper around PropertyMap to allow for easy and dynamic configurability of contract options.
+This includes things like the maximum number of years that a name can be registered for, etc.
+
+Anyone can read, but only admins can write, as all write methods are gated via permissions checks
+*/
+
+module aptos_names_v2::v2_config {
+    friend aptos_names_v2::v2_domains;
+
+    use aptos_framework::account;
+    use aptos_framework::aptos_account;
+    use std::error;
+    use std::signer;
+    use std::string::{Self, String};
+
+    const CONFIG_OBJECT_SEED: vector<u8> = b"ANS v2 config";
+
+    const DOMAIN_COLLECTION_NAME: vector<u8> = b"Aptos Domain Names V2";
+    const SUBDOMAIN_COLLECTION_NAME: vector<u8> = b"Aptos Subdomain Names V2";
+
+    /// Raised if the signer is not authorized to perform an action
+    const ENOT_AUTHORIZED: u64 = 1;
+    /// Raised if there is an invalid value for a configuration
+    const EINVALID_VALUE: u64 = 2;
+    /// Domain length is invalid, it must be at least 3 characters
+    const EINVALID_DOMAIN_LENGTH: u64 = 3;
+
+    const SECONDS_PER_YEAR: u64 = 60 * 60 * 24 * 365;
+    const SECONDS_PER_DAY: u64 = 60 * 60 * 24;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
```

Comment:
this probably should have been its own resource group, but I guess this work lol